src/main/java/com/avispl/symphony/dal/infrastructure/management/philips/huebridge/PhilipsHueDeviceCommunicator.java

```diff
@@ -0,0 +1,4123 @@
+		String name = stats.get(property + RoomsAndZonesControlEnum.NAME.getName());
+		String type = stats.get(property + RoomsAndZonesControlEnum.TYPE.getName());
+		List<String> id = new LinkedList<>();
+		List<Children> deviceIdList = new LinkedList<>();
+		for (String value : mapOfDevice.values()) {
+			Children children = new Children();
+			if (!PhilipsConstant.NONE.equals(value) && !StringUtils.isNullOrEmpty(value)) {
+				for (Entry<String, String> idDevice : mapOfNameAndId.entrySet()) {
+					if (!id.contains(idDevice.getValue()) && value.equals(idDevice.getKey())) {
+						children.setRid(idDevice.getValue());
+						children.setType(PhilipsConstant.LIGHT.toLowerCase(Locale.ROOT));
+						deviceIdList.add(children);
+						id.add(idDevice.getValue());
+						break;
+					}
+				}
+			}
+		}
+		if (StringUtils.isNullOrEmpty(name) || StringUtils.isNullOrEmpty(name.trim())) {
+			throw new ResourceNotReachableException("Error while creating Zone, Zone name can't empty or space");
+		}
+		if (PhilipsConstant.NONE.equals(type)) {
+			throw new ResourceNotReachableException("Error while creating Zone, Zone type can't empty");
+		}
+		if (isEditZone) {
+			String nameValue = property.substring(PhilipsConstant.ZONE.length() + 1, property.indexOf(PhilipsConstant.HASH));
+			if (!PhilipsConstant.CREATE_ZONE.equals(name) && !name.equals(nameValue)) {
+				isRoomAndZoneNameExisting(name, false);
+			} else if (!PhilipsConstant.CREATE_ROOM.equals(nameValue) && !name.equals(nameValue)) {
+				isRoomAndZoneNameExisting(name, false);
+			}
+		} else {
+			isRoomAndZoneNameExisting(name, false);
+		}
+		type = RoomTypeEnum.getValueOfRoomTypeEnumByName(type);
+		MetaData metaData = new MetaData();
+		metaData.setArchetype(type);
+		metaData.setName(name);
+		RoomAndZoneResponse response = new RoomAndZoneResponse();
+		response.setChildren(deviceIdList.toArray(new Children[0]));
+		response.setType(PhilipsConstant.ZONE.toLowerCase(Locale.ROOT));
+		response.setMetaData(metaData);
+		return response;
+	}
+
+	/**
+	 * Retrieve list zones
+	 */
+	private void retrieveZones() {
+		try {
+			RoomAndZoneWrapper zoneWrapper = this.doGet(PhilipsUtil.getMonitorURL(PhilipsURL.ZONES), RoomAndZoneWrapper.class);
+			if (zoneWrapper != null && zoneWrapper.getData() != null) {
+				Collections.addAll(zoneList, zoneWrapper.getData());
+				for (RoomAndZoneResponse zone : zoneList) {
+					if (zone.getChildren().length > 0) {
+						zoneNameAndIdMap.put(PhilipsConstant.ALL_DEVICE_IN_ZONE + PhilipsConstant.DASH + zone.getMetaData().getName(), zone.getId());
+					}
+				}
+				zoneNameAndIdMap.put(PhilipsConstant.NONE, PhilipsConstant.NONE);
+			} else {
+				throw new ResourceNotReachableException("List zones data is empty");
+			}
+		} catch (Exception e) {
+			String errorMessage = String.format("List Zones Data Retrieval-Error: %s with cause: %s", e.getMessage(), e.getCause().getMessage());
```

Comment:
is it possible to include some zone-specific information here?

Reply:
I fixed it